src/matchInchi.py

```diff
@@ -1,57 +1,34 @@
-
-from rdkit import Chem,DataStructs
-from src.downloadPubchemInfo import getPubchemInfo
-
-
-
-def compareInchiByFingerprint(met1:str, met2:str, method:str = "Dice") -> float:
-    ''' Simple function to compare based on fingerprints
-    @met1 - str of the first inchi string
-    @met2 - str of the second inchi string
-    @method - str defining the method to use. One of 'Tanimoto', 'Dice', 'Cosine', 'Sokal', 'Russel', 'Kulczynski', 'McConnaughey'.'''
-    
-    # correct and check method string if implemented in RDKit
-    method = method[0].upper() + method[1:].lower()
-    impl_methods = ['Tanimoto', 'Dice', 'Cosine', 'Sokal', 'Russel', 'Kulczynski', 'McConnaughey'] # "Tversky" is noted in the documentation, but throws an error
-    if method not in impl_methods:
-        raise ValueError(method+ " not implemented. Choose one of: " + ", ".join(impl_methods))
-
-    # do the conversion
-    m1 = Chem.MolFromInchi(met1)
-    m2 = Chem.MolFromInchi(met2)
-    fp1 = Chem.RDKFingerprint(m1)
-    fp2 = Chem.RDKFingerprint(m2)
-    sim = DataStructs.FingerprintSimilarity(fp1,fp2, metric = eval("DataStructs."+method+"Similarity"))
-    return(sim)
-
-def compareInchiByCSmile(met1:str, met2:str) -> bool:
-    '''Converts Inchi to canonical smile and compares the strings'''
-    # do the conversion
-    m1 = Chem.MolFromInchi(met1)
-    m2 = Chem.MolFromInchi(met2)
-    sm1 = Chem.MolToSmiles(m1)
-    sm2 = Chem.MolToSmiles(m2)
-    return(sm1 == sm2)
-
-
-def compareInchiByKey(met1:str, met2:str) -> bool:
-    '''Converts Inchi to Inchi-key and compares the hashes'''
-    # do the conversion
-    m1 = Chem.MolFromInchi(met1)
-    m2 = Chem.MolFromInchi(met2)
-    key1 = Chem.MolToInchiKey(m1)
-    key2 = Chem.MolToInchiKey(m2)
-    return(key1 == key2)
-
-def compareInchiByString(met1:str, met2:str) -> bool:
-    '''Converts Inchi to canonical Inchi and compares the strings'''
-    # do the conversion
-    m1 = Chem.MolFromInchi(met1)
-    m2 = Chem.MolFromInchi(met2)
-    key1 = Chem.MolToInchi(m1)
-    key2 = Chem.MolToInchi(m2)
-    return(key1 == key2)
+# 27.04.23
+
+from src.matchInchiRoutines import *
+from rdkit import Chem
+
+def matchInchi(inchi1:str, inchi2:str) -> tuple:
+    ''' A combination of different matching algorithms for the Inchi-Strings, which end in a simple yes/no answer, whether the molecules are the same'''
+
+    # check whether the molecules are differently charged, if so neutralize the charges
+    m1 = Chem.MolFromInchi(inchi1)
+    m2 = Chem.MolFromInchi(inchi2)
+    charge1 = Chem.GetFormalCharge(m1)
+    charge2 = Chem.GetFormalCharge(m2)
+    if charge1 != charge2:
+        m1 = NeutraliseCharges(m1)
+        m2 = NeutraliseCharges(m2)
+        inchi1 = Chem.MolToInchi(m1)
+        inchi2 = Chem.MolToInchi(m2)
+    chargeDiff = charge1 - charge2
+
+    # check the fingerprints - they should be the same
+    if compareInchiByFingerprint(inchi1,inchi2) != 1.0:
+        return(False,chargeDiff)
```

Comment:
Just a really minor comment:
It would be considered a better programming practice to avoid having multiple return statements in the same function. It is generally recommended to create a return variable at the beginning of the code (in this case smth like "comparison_result = (False, 0)" ) and then replace the 3 return statements with a nested if elif else statement that assigns the return value to "comparison_result", and finally return comparison result at the end. That makes code more maintainable (it is easier to add more cases without messing up with the original logic)